src/protocols/bdx/BdxTransferSession.h

```diff
@@ -80,12 +80,22 @@ class DLL_EXPORT TransferSession
         bool IsEof           = false;
     };
 
+    struct MessageTypeData
+    {
+        uint32_t ProtocolId = 0; // Should only be SecureChannel or BDX
```

Comment:
Why not store this as an actual protocol id, as opposed to `uint32_t`?

Reply:
When I originally wrote this header, I ran into compilation errors when I tried using non-trivial types in the structs included in the anonymous union.

That being said, I tried doing this change today and got it to compile locally, so I'm not sure what I did wrong last time. I agree with you that it's much simpler and cleaner to use `Protocols::Id`

Reply:
New commit has these changes